spec/app/ics-020-fungible-token-transfer/README.md

```diff
@@ -54,6 +55,19 @@ The ICS 20 token denominations are represented by the form `{ics20Port}/{ics20Ch
 
+The memo is not parsed by the ICS20 application and may be used for external users or higher-level middleware and smart contracts. If the memo is intended to be used within the state machine, it must be in JSON dictionary format with the relevant information keyed for the intended user.
+
+The `forwardingPath` of the packet data is the path to the final destination of the tokens after they reach the immediate destination chain specified by the packet's channel identifiers. The `forwardingPath` may be specified as a list of port/channel identifier tuples prefixed by `channels:` or a list of chain IDs that will be resolved with an on-chain registry prefixed by `chains:`. The `chainID` specification may be unsupported, in which case the implementation should return an error acknowledgement. A special keyword `origin` may be prefixed to the `forwardingPath` list to instruct the forwarding logic to first unwind the token path to the origin chain before continuing with forwarding. If the specified `forwardingPath` is supported, the implementation will forward the path along one hop forward and asynchronously write the acknowledgement once the forwarded packet has returned.
+
+Example valid `forwardingPath`s:
+
+- `channels:transfer/channel-3/transfer/channel-2/transfer/channel-300`: On each receiving chain, the implementation will pop a (srcPortID, srcChannelID) from the front of the list, resolve to a channel, and then forward the tokens along. Once the `forwardingPath` is empty, it will send to specified receiver which must be a valid address at the final destination.
+- `channels:origin/transfer/channel-4`: This `forwardingPath` is similar to the above, but it will first unwind the tokens to their origin chain before executing the logic described above.
+- `channels:origin`: This will simply unwind the tokens to their origin chain and send to the intended receiver.
+- `chains:cosmoshub/osmosis/juno`: On each receiving chain the implementation must first consult its local chain registry to resolve the chainID to a source port and source channel, then it will forward the tokens along that channel. Once the forwardingPath is empty, it will send to specified receiver which must be a valid address at the final destination.
+- `chains:origin/juno/osmosis`: Similar to the previous logic, it will first unwind to the origin before resolving the first chainID and forwarding along the returned channel to forward along the rest of the path.
```

Comment:
Sanity check: We don't have any guarantees here that the chains recognized as `juno` and `osmosis` are the same ones recognized as such in the source chain, right? ie.: what origin considers to be "juno" may not the the same as what the sender chain considers as juno.

On the osmosis registries we are considering validating the channels by sending a packet through a list of channels that gets forwarded back to osmosis, and seeing that it returns with the expected path.

Reply:
This is correct. We could theoretically resolve in a single place like a global CNS, but this would require the global CNS to hold the relative identifiers of all chains to all chains.

Using this feature as it is currently specified would require the assumption that all chains hold the same social consensus